### ISSUE TITLE
Add a layer for general interpolation

### ISSUE THREAD

**Issue by sayakpaul** [keras-team/keras-cv#317]
Currently, `keras.layers.UpSampling2D` supports interpolation (to higher resolutions) but lacks interpolation methods such as bicubic. It probably also lacks support for aligning corners:

> align_corners ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – Geometrically, we consider the pixels of the input and output as squares rather than points. If set to True, the input and output tensors are aligned by the center points of their corner pixels, preserving the values at the corner pixels. If set to False, the input and output tensors are aligned by the corner points of their corner pixels, and the interpolation uses edge value padding for out-of-boundary values, making this operation independent of input size when scale_factor is kept the same. This only has an effect when mode is 'linear', 'bilinear', 'bicubic' or 'trilinear'.

Generic interpolation is a sought-after usecase in vision. For example, when adapting the pre-trained positional embeddings in Vision Transformers to higher resolutions, they are interpolated using bicubic interpolation. Currently, it's implemented using `tf.image.resize` ([example](https://github.com/huggingface/transformers/blob/main/src/transformers/models/vit/modeling_tf_vit.py#L96)). 

The motivation behind a generic interpolation layer would be to support usecases like that. Currently, the closest reference implementation is: https://pytorch.org/docs/stable/generated/torch.nn.functional.interpolate.html.

**Comment by LukeWood** [comment]
The KerasCV team is attempting to narrow the package scope in order to more efficiently use our limited resources.  As such, we are only accepting contributions directly related to items on the 6 month roadmap for the package.

Feel free to read more on the roadmap:
https://github.com/keras-team/keras-cv/blob/master/.github/ROADMAP.md
